src/debian-packages/cjdns/Makefile

```diff
@@ -2,5 +2,64 @@
-	echo TODO
+PACKAGE_NAME = cjdns
+VERSION = 20.0
+DEBIAN_REVISION = 1
+MAINTAINER = Benedict Lau <benedict.lau@groundupworks.com>
+SOURCE_URL = https://github.com/cjdelisle/cjdns
+
+TAG = cjdns-v20
+ARCHS = amd64 armhf
+BUILD_DIR = /tmp/$(PACKAGE_NAME)
+OUTPUT_DIR = /vagrant/output/debian-packages
+
+# Package deb for a particular architecture
+#
+# $1 is the architecture
+define package_deb_arch
+	mkdir -p $(BUILD_DIR)/debian/$(1)
+	cp -r debian $(BUILD_DIR)/debian/$(1)/    # Copy deb package files
+	sed -i "s|__PACKAGE__|$(PACKAGE_NAME)|g" $(BUILD_DIR)/debian/$(1)/debian/DEBIAN/control    # Replace __PACKAGE__ macro in control file
+	sed -i "s|__VERSION__|$(VERSION)-$(DEBIAN_REVISION)|g" $(BUILD_DIR)/debian/$(1)/debian/DEBIAN/control    # Replace __VERSION__ macro in control file
+	sed -i "s|__ARCHITECTURE__|$(1)|g" $(BUILD_DIR)/debian/$(1)/debian/DEBIAN/control    # Replace __ARCHITECTURE__ macro in control file
+	sed -i "s|__MAINTAINER__|$(MAINTAINER)|g" $(BUILD_DIR)/debian/$(1)/debian/DEBIAN/control    # Replace __MAINTAINER__ macro in control file
+	sed -i "s|__HOMEPAGE__|$(SOURCE_URL)|g" $(BUILD_DIR)/debian/$(1)/debian/DEBIAN/control    # Replace __HOMEPAGE__ macro in control file
+	sed -i "s|__PACKAGE__|$(PACKAGE_NAME)|g" $(BUILD_DIR)/debian/$(1)/debian/usr/share/doc/$(PACKAGE_NAME)/copyright    # Replace __PACKAGE__ macro in copyright file
+	sed -i "s|__MAINTAINER__|$(MAINTAINER)|g" $(BUILD_DIR)/debian/$(1)/debian/usr/share/doc/$(PACKAGE_NAME)/copyright    # Replace __MAINTAINER__ macro in copyright file
+	sed -i "s|__SOURCE__|$(SOURCE_URL)|g" $(BUILD_DIR)/debian/$(1)/debian/usr/share/doc/$(PACKAGE_NAME)/copyright    # Replace __SOURCE__ macro in copyright file
+	cp $(BUILD_DIR)/bin/$(1)/cjdroute $(BUILD_DIR)/debian/$(1)/debian/usr/bin/    # Copy binary
+	mkdir -p $(BUILD_DIR)/debian/$(1)/debian/lib/systemd/system    # Make directories for systemd service files
+	cp $(BUILD_DIR)/src/contrib/systemd/* $(BUILD_DIR)/debian/$(1)/debian/lib/systemd/system/    # Copy systemd service files
+	cd $(BUILD_DIR)/debian/$(1); \
+	  dpkg-deb --build debian; \
+	  cp debian.deb $(OUTPUT_DIR)/$(PACKAGE_NAME)_$(VERSION)-$(DEBIAN_REVISION)_$(1).deb
+endef
+
+package_deb: build_bin_amd64 build_bin_armhf
+	$(foreach arch,$(ARCHS),$(call package_deb_arch,$(arch));)
+
+build_bin_amd64: download_source
+	mkdir -p $(BUILD_DIR)/bin/amd64
+	cd $(BUILD_DIR)/src; \
+	  ./clean; \
+	  ./do; \
+	  cp cjdroute $(BUILD_DIR)/bin/amd64/
+
+build_bin_armhf: mkdir_build
+	mkdir -p $(BUILD_DIR)/bin/armhf
+	cp -r docker $(BUILD_DIR)/
+	sed -i "s|__TAG__|$(TAG)|g" $(BUILD_DIR)/docker/Dockerfile    # Replace __TAG__ macro in Dockerfile
+	mkdir -p $(BUILD_DIR)/docker/build
+	cd $(BUILD_DIR)/docker; \
+	  sudo docker build -t cjdns-armv7hf .    # Build armv7hf docker container
+	  sudo docker run --memory="2g" --memory-swap="4g" -v /usr/bin/qemu-arm-static:/usr/bin/qemu-arm-static -v $(BUILD_DIR)/docker/build:/tmp/cjdns cjdns-armv7hf    # Build binary in emulated ARM docker
```

Comment:
Most of the time the compilation just hangs, and the container's `CPU %` and `MEM USAGE` in `docker stats` drop to zero. Not sure if related to memory but adding these seem to be the only way to get things working at all so far. cjdns should not need this much resource to compile though.